crates/compiler/builtins/bitcode/src/str.zig

```diff
@@ -811,13 +864,40 @@ pub fn strSplit(string: RocStr, delimiter: RocStr) callconv(.C) RocList {
     return list;
 }
 
+const Init = fn (slice_bytes: [*]u8, len: usize, ref_ptr: usize) RocStr;
+fn initFromSmallStr(slice_bytes: [*]u8, len: usize, _: usize) RocStr {
+    return RocStr.init(slice_bytes, len);
+}
+
+// The ref_ptr must already be shifted to be ready for storing in a seamless slice.
+fn initFromBigStr(slice_bytes: [*]u8, len: usize, ref_ptr: usize) RocStr {
+    // Here we can make seamless slices instead of copying to a new small str.
+    return RocStr{
+        .str_bytes = slice_bytes,
+        .str_len = len | SEAMLESS_SLICE_BIT,
+        .str_capacity = ref_ptr,
+    };
+}
+
+// TODO: relpace this with @qualCast or @constCast in future version of zig
```

Comment:
will 0.10 have this?

Reply:
I believe 0.10 will have `@qualCast` then in 0.11 it is split into `@constCast` and some other cast.